From be6b8456d1930d63ebd9b54d1cf8891318b61976 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 02:16:37 -0400
Subject: [iommu] vt-d: Fix reference count in iommu_prepare_isa

Message-id: <20140905021637.32574.81936.stgit@gir.stowe>
Patchwork-id: 91369
O-Subject: [RHEL7.1 PATCH 10/17] iommu/vt-d: Fix reference count in iommu_prepare_isa
Bugzilla: 1136537
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1136537
Upstream Status: 9b27e82d208b85ee9c3146693bb1e92f9ade4468

commit 9b27e82d208b85ee9c3146693bb1e92f9ade4468
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Tue May 20 20:37:52 2014 +0800

    iommu/vt-d: Fix reference count in iommu_prepare_isa

    Decrease the device reference count avoid memory leak.

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Joerg Roedel <jroedel@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 7119079..8263fee 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -2404,6 +2404,7 @@ static inline void iommu_prepare_isa(void)
   printk(KERN_ERR "IOMMU: Failed to create 0-16MiB identity map; "
          "floppy might not work\n");
 
+ pci_dev_put(pdev);
 }
 #else
 static inline void iommu_prepare_isa(void)
-- 
1.7.1